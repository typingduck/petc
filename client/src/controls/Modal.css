
.petc-modal-overlay {
  background-color: rgba(255, 255, 255, 0.3);
  bottom: 0;
  left: 0;
  position: fixed;
  right: 0;
  top: 0;
}
