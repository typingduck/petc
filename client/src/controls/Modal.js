import React from 'react'
import './Modal.css'

/**
 * Replaces React Modal with a a simple implementation because react modal
 * attaches elements in portals which makes tests harder.
 */
export default class Modal extends React.Component {
  render () {
    if (this.props.isOpen) {
      return (
        <div className='petc-modal-overlay' onClick={this.props.onRequestClose}>
          <div className={'petc-modal ' + this.props.className} onClick={stopPropagation} >
            {this.props.children}
          </div>
        </div>
      )
    } else {
      return null
    }
  }
}

function stopPropagation (ev) {
  ev.stopPropagation()
}
