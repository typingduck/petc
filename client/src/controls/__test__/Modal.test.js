/* global expect it jest */
import React from 'react'
import Renderer from 'react-test-renderer'

import Modal from '../Modal'

const props = {
  isOpen: true,
  className: 'custom-clss',
  onRequestClose: jest.fn()
}

it('renders modal', () => {
  const component = Renderer.create(
    <Modal {...props} >
      <div id='inner' />
    </Modal>
  )
  expect(component.toJSON()).toMatchSnapshot()
})

it('hides modal when not debugging', () => {
  const component = Renderer.create(<Modal isOpen={false} />)
  expect(component.toJSON()).toBeNull()
})

it('click overlay to hide', () => {
  const component = Renderer.create(<Modal {...props} />)
  const overlay = component.toJSON()
  overlay.props.onClick()
  expect(props.onRequestClose).toHaveBeenCalled()
})

it('click modal should not hide', () => {
  const component = Renderer.create(<Modal {...props} />)
  const modal = component.toJSON().children[0]
  const stopProp = jest.fn()
  modal.props.onClick({ stopPropagation: stopProp })
  expect(stopProp).toHaveBeenCalled()
})
