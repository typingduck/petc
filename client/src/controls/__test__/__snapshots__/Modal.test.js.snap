// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders modal 1`] = `
<div
  className="petc-modal-overlay"
  onClick={[Function]}
>
  <div
    className="petc-modal custom-clss"
    onClick={[Function]}
  >
    <div
      id="inner"
    />
  </div>
</div>
`;
